pipeline/validation-build.yaml

```diff
@@ -44,3 +68,25 @@ jobs:
 
       - script: node $(system.defaultWorkingDirectory)/pipeline/verify-clean-lockfile.js
         displayName: verify that building did not change the lockfile unexpectedly
+
+  - job: 'e2e_test'
+
+    dependsOn: 'gradlew_assemble'
+
+    pool:
+        # per ADO docs, only macOS hosted agents support Android emulators. The emulator is installed on other agents, but didn't 
+        # function properly on Ubuntu when this was added.
+        # https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/android?view=azure-devops#test-on-the-android-emulator
+        vmImage: 'macOS-latest' 
```

Comment:
macOS makes sense here given the constraint. I noticed the other jobs are `ubuntu` and `windows`. As a reader I'm not sure if that is intentional or happenstance. Is there a reason we prefer those OS types for the particular jobs?

Reply:
We currently use Windows, which I left as-is. I noticed the build occurs faster on Ubuntu, so I added the new job in Ubuntu. Perhaps I should just switch the others from Windows to Ubuntu as well? Not sure if there are any side effects...

Reply:
I think switching to the fastest is worth a try! If we find out that Windows is required that'll be nice so we can document it

Reply:
Switched to Ubuntu!